### DISCUSSION TITLE
Cannot read GML geometry because of missing DOMParser in node

### DISCUSSION THREAD

**Discussion by dodinh** [openlayers/openlayers#16297] [Q&A] [answered]
I am trying to use openlayers to read a GML geometry in nodejs. However, it fails because internally, ol uses DOMParser which is only available in the browser but not in node ("DOMParser is not defined"). Where is the problem in my thought process, and how would I parse GML correctly?

```
async function readGML(inputGeometry) {
    let GML32 = (await import('ol/format/GML32.js')).default;
    let geometry = new GML32().readGeometry(inputGeometry);
    return geometry;
}
```

Using ol v10, node v20.

**Comment by ahocevar** [comment, marked as answer]
You can use https://www.npmjs.com/package/@xmldom/xmldom, and add it to the global scope. Something like
```js
import { DOMParser, XMLSerializer } from '@xmldom/xmldom';

globalThis.DOMParser = DOMParser;
globalThis.XMLSerializer = XMLSerializer;
